libtransmission/peer-mgr.cc

```diff
@@ -797,6 +799,32 @@ class tr_swarm
 
 struct tr_peerMgr
 {
+private:
+    static auto constexpr BandwidthTimerPeriod = 500ms;
+    static auto constexpr RechokePeriod = 10s;
+    static auto constexpr RefillUpkeepPeriod = 10s;
+
+    // Max number of outbound peer connections to initiate.
+    // This throttle is an arbitrary number to avoid overloading routers.
+    static auto constexpr MaxConnectionsPerSecond = size_t{ 18U };
+    static auto constexpr MaxConnectionsPerPulse = size_t(MaxConnectionsPerSecond * BandwidthTimerPeriod / 1s);
+
+    // Building a peer candidate list is expensive, so cache it across pulses.
+    // We want to cache it long enough to avoid excess CPU cycles,
+    // but short enough that the data isn't too stale.
+    static auto constexpr OutboundCandidatesListTtl = 2s;
```

Comment:
Actually, maybe this value should be defined as `BandwidthTimerPeriod * 4`, because you are targeting a 75% decrease in CPU time compared to the existing implementation.

Reply:
That is clearer. Updated in 2d1a45904ad5103ddfd9536ef48fbad8299daf6f